lib/std/os/uefi/tables/runtime_services.zig

```diff
@@ -23,50 +25,549 @@ pub const RuntimeServices = extern struct {
+    pub fn getVariableSize(
+        self: *const RuntimeServices,
+        name: [*:0]const u16,
+        guid: *align(8) const Guid,
+    ) GetVariableSizeError!?struct { usize, VariableAttributes } {
+        var size: usize = 0;
+        var attrs: VariableAttributes = undefined;
+
+        switch (self._getVariable(
+            name,
+            guid,
+            &attrs,
+            &size,
+            null,
+        )) {
+            .success, .buffer_too_small => return .{ size, attrs },
+            .not_found => return null,
+            .device_error => return Error.DeviceError,
+            .unsupported => return Error.Unsupported,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
+    }
+
+    /// To determine the minimum necessary buffer size for the variable, call
+    /// `getVariableSize` first.
+    pub fn getVariable(
+        self: *const RuntimeServices,
+        name: [*:0]const u16,
+        guid: *align(8) const Guid,
+        buffer: []u8,
+    ) GetVariableError!?struct { []u8, VariableAttributes } {
+        var attrs: VariableAttributes = undefined;
+        var len = buffer.len;
+
+        switch (self._getVariable(
+            name,
+            guid,
+            &attrs,
+            &len,
+            buffer.ptr,
+        )) {
+            .success => return .{ buffer[0..len], attrs },
+            .not_found => return null,
+            .buffer_too_small => return Error.BufferTooSmall,
+            .device_error => return Error.DeviceError,
+            .unsupported => return Error.Unsupported,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
+    }
+
+    pub fn variableNameIterator(
+        self: *const RuntimeServices,
+        buffer: []u16,
+    ) VariableNameIterator {
+        buffer[0] = 0;
+        return .{
+            .services = self,
+            .buffer = buffer,
+            .guid = undefined,
+        };
+    }
+
+    pub fn setVariable(
+        self: *RuntimeServices,
+        name: [*:0]const u16,
+        guid: *align(8) const Guid,
+        attributes: VariableAttributes,
+        data: []const u8,
+    ) SetVariableError!void {
+        switch (self._setVariable(
+            name,
+            guid,
+            attributes,
+            data.len,
+            data.ptr,
+        )) {
+            .success => {},
+            .invalid_parameter => return Error.InvalidParameter,
+            .out_of_resources => return Error.OutOfResources,
+            .device_error => return Error.DeviceError,
+            .write_protected => return Error.WriteProtected,
+            .security_violation => return Error.SecurityViolation,
+            .not_found => return Error.NotFound,
+            .unsupported => return Error.Unsupported,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
+    }
+
+    pub fn getNextHighMonotonicCount(
+        self: *const RuntimeServices,
+        count: *u32,
```

Comment:
This is not consistent with the rest of the interface you have, this should be returned instead of being an out-parameter.